pkg/pillar/docs/watcher.md

```diff
@@ -0,0 +1,92 @@
+# Watcher
+
+The **watcher** component is the part of our system responsible for
+monitoring and managing system resources to ensure optimal performance and
+stability. It oversees various aspects such as memory usage, disk space, and the
+number of active goroutines. By keeping track of these metrics, the watcher can
+detect anomalies, trigger garbage collection, and alert us to potential issues
+like resource exhaustion or leaks.
+
+## Manual Garbage Collection
```

Comment:
In addition to these recent additions, it would make sense to also document the original logging around HostMemory and DiskUsage. Maybe @gkodali-zededa can propose a paragraph or two?

Reply:
It would be nice, but I needed to study this part more to document it in the scope of this PR...